src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -27,8 +27,10 @@ public enum Flags
             NamedFormulas = 1 << 1
         }
 
+        private bool _hasSemicolon = false;
+
         private readonly TokenCursor _curs;
-        private readonly Flags _flags;
+        private Flags _flags;
```

Comment:
This being mutable is a "red flag". Should we be using a stack here instead?

Reply:
Changed